exist-core/src/main/java/org/exist/xquery/functions/xmldb/FunXCollection.java

```diff
@@ -21,37 +21,48 @@
  */
 package org.exist.xquery.functions.xmldb;
 
-//import org.apache.logging.log4j.LogManager;
-
-import org.exist.dom.QName;
-import org.exist.xquery.Cardinality;
 import org.exist.xquery.FunctionSignature;
+import org.exist.xquery.XPathException;
 import org.exist.xquery.XQueryContext;
 import org.exist.xquery.functions.fn.ExtCollection;
-import org.exist.xquery.value.FunctionReturnSequenceType;
-import org.exist.xquery.value.FunctionParameterSequenceType;
-import org.exist.xquery.value.SequenceType;
+import org.exist.xquery.value.Sequence;
 import org.exist.xquery.value.Type;
 
+import java.net.URI;
+
+import static org.exist.xquery.FunctionDSL.*;
+import static org.exist.xquery.functions.xmldb.XMLDBModule.functionSignature;
+
 /**
- * @author wolf
- * @author ljo
+ * @author <a href="mailto:adam@evolvedbinary.com">Adam Retter</a>
  */
 public class FunXCollection extends ExtCollection {
-//    private static final Logger logger = LogManager.getLogger(FunXCollection.class);
-	public final static FunctionSignature signature =
-        new FunctionSignature(
-            new QName("xcollection", XMLDBModule.NAMESPACE_URI, XMLDBModule.PREFIX),
-            "Returns the document nodes in the collections $collection-uris " +
-            "non-recursively, i.e. does not include document nodes found in " +
-            "sub-collections. " +
-            "C.f. fn:collection(). " +
-            XMLDBModule.COLLECTION_URI,
-            new SequenceType[] {
-		new FunctionParameterSequenceType("collection-uris", Type.STRING, Cardinality.ONE_OR_MORE, "The collection URIs")},
-            new FunctionReturnSequenceType(Type.NODE, Cardinality.ZERO_OR_MORE, "the document nodes from the specified collections excluding sub-collections"));
-
-	public FunXCollection(XQueryContext context) {
-		super(context, signature, false);
+
+	private static final String FS_XCOLLECTION_NAME = "xcollection";
+	static final FunctionSignature FS_XCOLLECTION = functionSignature(
+			FS_XCOLLECTION_NAME,
+			"Returns the document nodes in the collections $collection-uris " +
```

Comment:
mismatch of argument name in description and signature
`collection-uris` <> `arg`
This function is not defined in the XQ31 spec but part of the xmldb namespace.

Reply:
Ah! I see that I had some outstanding stashed changes that I had not pushed. I have now pushed those so that the `xmldb:xcollection` function is re-aligned with the updated `fn:collection` function.